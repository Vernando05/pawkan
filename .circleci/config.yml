commands:
  create_cert:
    description: |
      Create certificates
    steps:
      - run:
          name: "Create certificates"
          command: |
            echo "$SERVERCRT" | base64 -d >> server.crt
            echo "$SERVERKEY" | base64 -d >> server.key
  build:
    description: |
      Create a production build
    steps:
      - run:
          name: "Create a production build"
          command: |
            NODE_ENV=production npm run build
            tar -cvzf build_artifacts.tar.gz dist
  test:
    description: |
      Test using Jest and Cypress
    steps:
      - run:
          name: "Jest unit testing"
          command: |
            npm run test:unit -- --collectCoverage
      - run: 
          name: "Cypress e2e testing"
          command: npm run test-ci
      - codecov/upload:
          file: coverage/clover.xml
  deploy:
    description: |
      Deploy to Firebase hosting
    steps:    
      - run: ./node_modules/.bin/firebase deploy --token "$FIREBASE_TOKEN" --only hosting
  lint:
    description: |
      Lint
    steps:
      - run:
          name: "Lint files"
          command: npm run lint
  install:
    description: |
      Install dependencies
    steps:
      - run:
          name: "Install dependencies"
          command: npm ci
orbs:
  codecov: codecov/codecov@1.1.3
executors:
  default:
    description: |
      Node 12 runtime environment
    docker:
      - image: cypress/browsers:<<parameters.cypress>>
    parameters:
      cypress:
        default: node12.13.0-chrome78-ff70
        type: string
jobs:
  build:
    description: |
      Build the application in production mode. The built assets are stored in artifacts.
    executor: default
    steps:
      - attach_workspace:
          at: ~/
      - create_cert
      - build
      - persist_to_workspace:
          paths:
            - project
          root: ~/
      - store_artifacts:
          path: ./build_artifacts.tar
  test:
    description: |
      Run the tests and generate coverage reports.
    executor: default
    steps:
      - attach_workspace:
          at: ~/
      - create_cert
      - test
  deploy:
    description: |
      Deploy to production
    executor: default
    steps:
      - attach_workspace:
          at: ~/
      - deploy
  lint:
    description: |
      Run lint on the source code
    executor: default
    steps:
      - attach_workspace:
          at: ~/
      - lint
  install:
    description: |
      Install dependencies. Checkout is called before install. (Optional) To prevent checkout from being called, pass your steps to before_install.
    executor: default
    parameters:
      before_install:
        default:
          - checkout
        description: |
          Steps to be executed before restoring cache and running install. Default is to run 'checkout' command.
        type: steps
    steps:
      - steps: <<parameters.before_install>>
      - restore_cache:
          key: v1-node-modules-{{ checksum "package-lock.json" }}
      - install
      - persist_to_workspace:
          paths:
            - project
            - .cache/Cypress
          root: ~/
      - save_cache:
          key: v1-node-modules-{{ checksum "package-lock.json" }}
          paths:
            - ~/.npm
            - ~/.cache
version: 2.1
workflows:
  version: 2.1
  front-test-build-deploy:
    jobs:
      - install
      - test:
          requires:
            - install
      - build:
          filters:
            branches:
              only:
                - dev
          requires:
            - test
      - deploy:
          filters:
            branches:
              only:
                - dev
          requires:
            - build
