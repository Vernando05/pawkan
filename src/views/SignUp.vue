<template>
  <div>
    <FormProfile ref="formProfile" :submit="submit" :error="errorMessages" />
  </div>
</template>

<script lang="ts">
  import store from '@/store'
  import mixins from 'vue-typed-mixins'  
  import { mapGetters, mapMutations } from 'vuex'
  import FormProfile from '@/components/FormProfile.vue'
  import signUp from '@/mixins/signUp'
  import signIn from '@/mixins/signIn'
  import clientStorageResetToken from '@/mixins/clientStorageResetToken'
  import { CustomerCreateInput, CustomerUserError } from '@/types/shopify-storefront'

  export default mixins(signUp, signIn, clientStorageResetToken).extend ({
    name: 'SignUp',
    components: {
      FormProfile
    },
    metaInfo () {
      return {
        title: this.$t('signUp'),
        titleTemplate: '%s | Pawkan - Your best pet care'      
      }    
    },
    data: () => ({
      errorMessages: [] as CustomerUserError[]
    }),
    computed: {
      ...mapGetters([
        'prevNav'
      ])
    },
    methods: {
      ...mapMutations([
        'setCustomerAccessToken'
      ]),
      async submit (inputCustomerCreate: CustomerCreateInput): Promise<void> {
        const { email, password } = inputCustomerCreate
        const inputCustomerAccessTokenCreate =  { email, password }
        const result = await this.signUp_createUser(inputCustomerCreate, inputCustomerAccessTokenCreate)
        if (result.customerUserErrors.length) {
          this.errorMessages = result.customerUserErrors
        } else {
          const result = await this.signIn_createToken(inputCustomerAccessTokenCreate)                    
          if (result.customerUserErrors.length) {
            this.errorMessages = result.customerUserErrors
          } else
          if (result.customerAccessToken) {
            (this.$refs.formProfile as InstanceType<typeof FormProfile>).reset()
            this.setCustomerAccessToken(result.customerAccessToken.accessToken)
            this.clientStorageResetToken_reset(result.customerAccessToken.accessToken)    
            this.$router.push(this.prevNav)
          }
        }       
      }
    },
    beforeRouteEnter (...[, , next]) {
      if (store.getters.customerAccessToken) {
        next({ name: 'Home', params: { lang: 'en' } })
      } else {
        next()
      }
    }
  })
</script>