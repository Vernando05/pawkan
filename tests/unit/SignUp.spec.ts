import { renderComponent, flush } from './config'
import SignUp from '@/views/SignUp.vue'
import { fireEvent } from '@testing-library/vue'
import snapshotDiff  from 'snapshot-diff'
import Vue from 'vue'
import { VueTelInput } from 'vue-tel-input'
require('es6-promise').polyfill()
require('isomorphic-fetch')

jest.useFakeTimers()

test('user filling the signup form', async () => {
  const store = {
    getters: {
      prevNav: jest.fn()
    }
  }
  const { container, getByRole, getByLabelText, queryByText } = renderComponent(SignUp, {
    store
  }, () => Vue.use(VueTelInput, { disabledFetchingCountry: true, defaultCountry: 'id' } as any))

  await flush()
  const pristine = container.cloneNode(true)
  expect(getByRole('button', { name: /submit/i })).toBeDisabled()

  // field first name
  expect(queryByText(/empty/i)).not.toBeInTheDocument()
  await fireEvent.touch(getByRole('textbox', { name: /first name/i }))
  await flush()
  expect(queryByText(/empty/i)).toBeInTheDocument()

  await fireEvent.update(getByRole('textbox', { name: /first name/i }), 'jack who climb a giant peanut tree with his strong arms without even breaking a sweat')
  await flush()
  expect(queryByText(/empty/i)).not.toBeInTheDocument()
  expect(queryByText(/greater /i)).toBeInTheDocument()
  expect(getByRole('button', { name: /submit/i })).toBeDisabled()

  await fireEvent.update(getByRole('textbox', { name: /first name/i }), 'jack')
  await flush()
  expect(queryByText(/greater /i)).not.toBeInTheDocument()
  expect(getByRole('button', { name: /submit/i })).toBeDisabled()

  // field email
  expect(queryByText(/empty/i)).not.toBeInTheDocument()
  await fireEvent.touch(getByRole('textbox', { name: /e-mail/i }))
  await flush()
  expect(queryByText(/empty/i)).toBeInTheDocument()
  expect(getByRole('button', { name: /submit/i })).toBeDisabled()

  await fireEvent.update(getByRole('textbox', { name: /e-mail/i }), 'jack@lumber')
  await flush()
  expect(queryByText(/empty/i)).not.toBeInTheDocument()
  expect(queryByText(/be valid/i)).toBeInTheDocument()
  expect(getByRole('button', { name: /submit/i })).toBeDisabled()

  await fireEvent.update(getByRole('textbox', { name: /e-mail/i }), 'jack@lumber.com')
  await flush()
  expect(queryByText(/be valid/i)).not.toBeInTheDocument()
  expect(getByRole('button', { name: /submit/i })).toBeDisabled()

  // field mobile
  await fireEvent.touch(getByLabelText(/mobile/i))
  expect(queryByText(/empty/i)).toBeInTheDocument()

  await fireEvent.update(getByLabelText(/mobile/i), '+628123456')
  expect(queryByText(/empty/i)).not.toBeInTheDocument()
  expect(queryByText(/invalid/i)).toBeInTheDocument()

  await fireEvent.update(getByLabelText(/mobile/i), '+628123456789')
  expect(queryByText(/invalid/i)).not.toBeInTheDocument()

  // field password
  expect(queryByText(/empty/i)).not.toBeInTheDocument()
  await fireEvent.touch(getByLabelText(/^password$/i))
  await flush()
  expect(queryByText(/empty/i)).toBeInTheDocument()
  expect(getByRole('button', { name: /submit/i })).toBeDisabled()

  await fireEvent.update(getByLabelText(/^password$/i), 'ichop')
  await flush()
  expect(queryByText(/empty/i)).not.toBeInTheDocument()
  expect(queryByText(/at least/i)).toBeInTheDocument()
  expect(getByRole('button', { name: /submit/i })).toBeDisabled()

  await fireEvent.update(getByLabelText(/^password$/i), 'ichoptreeusingaxe')
  await flush()
  expect(queryByText(/at least/i)).not.toBeInTheDocument()
  expect(queryByText(/greater /i)).toBeInTheDocument()
  expect(getByRole('button', { name: /submit/i })).toBeDisabled()

  await fireEvent.update(getByLabelText(/^password$/i), 'ichoptree')
  await flush()
  expect(queryByText(/greater/i)).not.toBeInTheDocument()
  expect(getByRole('button', { name: /submit/i })).toBeDisabled()

  // field confirmation email
  await fireEvent.update(getByLabelText(/confirm password/i),'ichoptreee')
  await fireEvent.touch(getByLabelText(/confirm password/i))
  await flush()
  expect(queryByText(/must be same/i)).toBeInTheDocument()

  await fireEvent.update(getByLabelText(/confirm password/i), 'ichoptree')
  await fireEvent.touch(getByLabelText(/confirm password/i))
  await flush()
  expect(queryByText(/must be same/i)).not.toBeInTheDocument()

  expect(getByRole('button', { name: /submit/i })).toBeEnabled()

  expect(snapshotDiff(pristine, container.cloneNode(true))).toMatchSnapshot()
}, 30000)