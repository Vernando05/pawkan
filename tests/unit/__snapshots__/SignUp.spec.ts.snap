// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user filling the signup form 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -7,11 +7,11 @@
         
        <span>
          <form>
            <span>
              <div
-               class=\\"v-input theme--light v-text-field\\"
+               class=\\"v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted\\"
              >
                <div
                  class=\\"v-input__control\\"
                >
                  <div
@@ -19,11 +19,11 @@
                  >
                    <div
                      class=\\"v-text-field__slot\\"
                    >
                      <label
-                       class=\\"v-label theme--light\\"
+                       class=\\"v-label v-label--active theme--light\\"
                        for=\\"input-11\\"
                        style=\\"left: 0px; position: absolute;\\"
                      >
                        First name
                      </label>
@@ -47,20 +47,20 @@
                      />
                    </div>
                    <div
                      class=\\"v-counter theme--light\\"
                    >
-                     0 / 30
+                     4 / 30
                    </div>
                  </div>
                </div>
              </div>
            </span>
             
            <span>
              <div
-               class=\\"v-input theme--light v-text-field\\"
+               class=\\"v-input theme--light v-text-field v-text-field--is-booted\\"
              >
                <div
                  class=\\"v-input__control\\"
                >
                  <div
@@ -105,11 +105,11 @@
              </div>
            </span>
             
            <span>
              <div
-               class=\\"v-input theme--light v-text-field\\"
+               class=\\"v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted\\"
              >
                <div
                  class=\\"v-input__control\\"
                >
                  <div
@@ -117,11 +117,11 @@
                  >
                    <div
                      class=\\"v-text-field__slot\\"
                    >
                      <label
-                       class=\\"v-label theme--light\\"
+                       class=\\"v-label v-label--active theme--light\\"
                        for=\\"input-19\\"
                        style=\\"left: 0px; position: absolute;\\"
                      >
                        E-mail
                      </label>
@@ -168,11 +168,11 @@
                >
                  <span
                    class=\\"vti__selection\\"
                  >
                    <div
-                     class=\\"vti__flag\\"
+                     class=\\"vti__flag id\\"
                    />
                     
                    <!---->
                     
                    <span
@@ -4341,11 +4341,11 @@
                  autocomplete=\\"on\\"
                  class=\\"vti__input\\"
                  id=\\"telInput0\\"
                  maxlength=\\"25\\"
                  name=\\"telephone\\"
-                 placeholder=\\"\\"
+                 placeholder=\\"+62 812-345-678\\"
                  tabindex=\\"0\\"
                  type=\\"tel\\"
                />
              </div>
               
@@ -4356,11 +4356,11 @@
              </p>
            </div>
             
            <span>
              <div
-               class=\\"v-input theme--light v-text-field\\"
+               class=\\"v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted\\"
              >
                <div
                  class=\\"v-input__control\\"
                >
                  <div
@@ -4368,11 +4368,11 @@
                  >
                    <div
                      class=\\"v-text-field__slot\\"
                    >
                      <label
-                       class=\\"v-label theme--light\\"
+                       class=\\"v-label v-label--active theme--light\\"
                        for=\\"input-25\\"
                        style=\\"left: 0px; position: absolute;\\"
                      >
                        Password
                      </label>
@@ -4397,20 +4397,20 @@
                      />
                    </div>
                    <div
                      class=\\"v-counter theme--light\\"
                    >
-                     0 / 10
+                     9 / 10
                    </div>
                  </div>
                </div>
              </div>
            </span>
             
            <span>
              <div
-               class=\\"v-input theme--light v-text-field\\"
+               class=\\"v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted\\"
              >
                <div
                  class=\\"v-input__control\\"
                >
                  <div
@@ -4418,11 +4418,11 @@
                  >
                    <div
                      class=\\"v-text-field__slot\\"
                    >
                      <label
-                       class=\\"v-label theme--light\\"
+                       class=\\"v-label v-label--active theme--light\\"
                        for=\\"input-29\\"
                        style=\\"left: 0px; position: absolute;\\"
                      >
                        Confirm Password
                      </label>
@@ -4447,11 +4447,11 @@
                      />
                    </div>
                    <div
                      class=\\"v-counter theme--light\\"
                    >
-                     0 / 10
+                     9 / 10
                    </div>
                  </div>
                </div>
              </div>
            </span>
@@ -4504,12 +4504,11 @@
                </div>
              </div>
            </span>
             
            <button
-             class=\\"mr-4 v-btn v-btn--disabled v-btn--has-bg theme--light v-size--default\\"
-             disabled=\\"disabled\\"
+             class=\\"mr-4 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default\\"
              type=\\"submit\\"
            >
              <span
                class=\\"v-btn__content\\"
              >"
`;
